src/main/CommandHandler.cpp

```diff
@@ -364,36 +364,27 @@ enum class Requirement
 };
 
 template <typename T>
-bool
+optional<T>
 parseNumParam(std::map<std::string, std::string> const& map,
-              std::string const& key, T& val, std::string& retStr,
-              Requirement requirement, bool& valueUpdated)
+              std::string const& key, T& val)
```

Comment:
you don't need to pass in `val` anymore.
If people want to do default values, I would expect callers to do things like
```
int n = 1000; // default value
auto v = parseNumParam<int>(m, "n");
if (v)
{
    n = *v;
}
```
non optional case is simply:
```
int n = *parseNumParam<int>(m, "n");
```